cl/search/tasks.py

```diff
@@ -83,26 +85,28 @@ def add_or_update_recap_docket(
     if data is None:
         return
 
-    si = scorched.SolrInterface(settings.SOLR_RECAP_URL, mode="w")
-    some_time_ago = now() - timedelta(seconds=update_threshold)
-    d = Docket.objects.get(pk=data["docket_pk"])
-    too_fresh = d.date_last_index is not None and (
-        d.date_last_index > some_time_ago
-    )
-    update_not_required = not data.get("content_updated", False)
-    if all([too_fresh, update_not_required]):
-        return
-    else:
-        try:
-            si.add(d.as_search_list())
-            if force_commit:
-                si.commit()
-            si.conn.http_connection.close()
-        except SolrError as exc:
-            add_or_update_recap_docket.retry(exc=exc, countdown=30)
+    with Session() as session:
+        si = scorched.SolrInterface(
+            settings.SOLR_RECAP_URL, http_connection=session, mode="w"
+        )
+        some_time_ago = now() - timedelta(seconds=update_threshold)
+        d = Docket.objects.get(pk=data["docket_pk"])
+        too_fresh = d.date_last_index is not None and (
+            d.date_last_index > some_time_ago
+        )
+        update_not_required = not data.get("content_updated", False)
+        if all([too_fresh, update_not_required]):
+            return
         else:
-            d.date_last_index = now()
-            d.save()
+            try:
+                si.add(d.as_search_list())
+                if force_commit:
+                    si.commit()
```

Comment:
I appreciate that you didn't refactor, but this feels like a good opportunity to just put these couple of Solr things inside the context block, instead of this whole thing?

Reply:
I'd say refactoring is best left for a follow up. I'm not super familiar with the codebase so was trying to keep the changes here as safe as possible.